cylc/flow/workflow_files.py

```diff
@@ -1235,32 +1233,30 @@ def get_run_dir_info(
     return relink, run_num, rundir
 
 
-def detect_flow_exists(run_path_base, numbered):
+def detect_flow_exists(
+    run_path_base: Union[Path, str], numbered: bool
+) -> bool:
     """Returns True if installed flow already exists.
 
     Args:
-        run_path_base (Path):
-            Workflow run directory i.e ~/cylc-run/<flow_name>
-        numbered (bool):
-            If true, will detect if numbered runs exist
-            If false, will detect if non-numbered runs exist, i.e. runs
-            installed by --run-name)
-
-    Returns:
-        True if installed flows exist.
-
+        run_path_base: Absolute path of workflow directory,
+            i.e ~/cylc-run/<flow_name>
+        numbered: If True, will detect if numbered runs exist. If False, will
+            detect if non-numbered runs exist, i.e. runs installed
+            by --run-name.
     """
     for entry in Path(run_path_base).iterdir():
-        isNumbered = bool(re.search(r'^run\d+$', entry.name))
+        is_numbered = bool(re.search(r'^run\d+$', entry.name))
         if (
             entry.is_dir()
-            and entry.name not in [
+            and entry.name not in {
                 WorkflowFiles.Install.DIRNAME, WorkflowFiles.RUN_N
-            ]
+            }
```

Comment:
Something I recently learned is that checking for membership in a set is faster than checking in a list. (Not that it will realistically make any difference in this case)